library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -363,6 +419,16 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
         continue;
       }
 
+      if (isCodeForUnsupportedMode(ccData1, ccData2)) {
```

Comment:
If I'm not mistaken, it seems that this code can move to handleCrtl? That is probably a better place to handle these transitions as it's part of "control code handling" in general.

Reply:
I do not think so. We need stop processing all data (characters included) in case we are not in CAPTION mode. Moving this down would add the characters of the TEXT and XDS modes to the buffers of our processing of CAPTION mode.